lib/erb_lint/reporters/junit_reporter.rb

```diff
@@ -0,0 +1,112 @@
+      end
+
+      private
+
+      CONTEXT = {
+        prologue_quote: :quote,
+        attribute_quote: :quote,
+      }
+
+      def create_junit_xml
+        # create prologue
+        xml = REXML::Document.new(nil, CONTEXT)
+        xml << REXML::XMLDecl.new("1.0", "UTF-8")
+
+        xml.add_element(create_testsuite_element)
+
+        xml
+      end
+
+      def create_testsuite_element
+        tests = stats.processed_files.size
+        failures = stats.found
+        testsuite_element = REXML::Element.new("testsuite", nil, CONTEXT)
+        testsuite_element.add_attribute("name", "erblint")
+        testsuite_element.add_attribute("tests", tests.to_s)
+        testsuite_element.add_attribute("failures", failures.to_s)
+
+        testsuite_element.add_element(create_properties)
+
+        processed_files.each do |filename, offenses|
+          if offenses.empty?
+            testcase_element = REXML::Element.new("testcase", nil, CONTEXT)
+            testcase_element.add_attribute("name", filename.to_s)
+            testcase_element.add_attribute("file", filename.to_s)
+
+            testsuite_element.add_element(testcase_element)
+          end
+
+          offenses.each do |offense|
+            testsuite_element.add_element(create_testcase(filename, offense))
+          end
+        end
+
+        testsuite_element
+      end
+
+      def create_properties
+        properties_element = REXML::Element.new("properties", nil, CONTEXT)
+
+        [
+          ["erb_lint_version", ERBLint::VERSION],
+          ["ruby_engine", RUBY_ENGINE],
+          ["ruby_version", RUBY_VERSION],
+          ["ruby_patchlevel", RUBY_PATCHLEVEL.to_s],
+          ["ruby_platform", RUBY_PLATFORM],
+        ].each do |property_attribute|
+          properties_element.add_element(create_property(*property_attribute))
+        end
+
+        properties_element
+      end
+
+      def create_property(name, value)
+        property_element = REXML::Element.new("property")
+        property_element.add_attribute("name", name)
+        property_element.add_attribute("value", value)
+
+        property_element
+      end
+
+      def create_testcase(filename, offense)
+        testcase_element = REXML::Element.new("testcase", nil, CONTEXT)
+        testcase_element.add_attribute("name", filename.to_s)
+        testcase_element.add_attribute("file", filename.to_s)
+        testcase_element.add_attribute("lineno", offense.line_number.to_s)
+
+        testcase_element.add_element(create_failure(filename, offense))
+
+        testcase_element
+      end
+
+      def create_failure(filename, offense)
+        message = offense.message
+        type = offense.simple_name
+
+        failure_element = REXML::Element.new("failure", nil, CONTEXT)
+        failure_element.add_attribute("message", "#{type}: #{message}")
+        failure_element.add_attribute("type", type.to_s)
+
+        cdata_element = REXML::CData.new("#{filename}:#{offense.line_number}:#{offense.column}")
```

Comment:
Can we add more information here like the error message?

The way we build Buildkite annotations ignores the `message` attribute if there's text content in the element:

<img width="1092" alt="image" src="https://github.com/Shopify/erb-lint/assets/3535/d45879b9-e2d9-45fe-8567-98e3bf1ad4c7">

We could do a better job on our end to handle that, but it still feels like the text content could be more comprehensive.

#296 was using the message in the text of the element: https://github.com/Shopify/erb-lint/pull/296/files#diff-77e0e3207b9a621523ef25749ac60fce67a306eeb27715133264667ffbf2cbd4R14
ci-queue also has the more information than the location, the message and backtrace cleaned up a little bit; https://github.com/Shopify/ci-queue/blob/941e4fe39a5bfdfc3bb8c0ec2c5a0dfa71075c3e/ruby/lib/minitest/queue/junit_reporter.rb#L118

Reply:
Ok, I will fix the failure tag's text content to contain the message attribute.

Example: `app/views/subscriptions/_loader.html.erb:1:7` to `SpaceInHtmlTag: Extra space detected where there should be no space. at app/views/subscriptions/_loader.html.erb:1:7`.

Reply:
@etiennebarrie 
I fixed by commit: 2fe02bb88860680c9a60c8785f40fe1a87876c5b.

Reply:
I forgot fix test...
please wait.

Reply:
@etiennebarrie 
Sorry, I re-committed: 9ef15e20da0ad46077c88b73bac06e4edd15d2c2